src/Tinker.jl

```diff
@@ -12,24 +12,40 @@ struct Rectangle <: Shape
     y::Number
     w::Number
     h::Number
+    pts::AbstractArray
```

Comment:
You should be able to use dispatch to specialize drawing. E.g.,
```julia
function draw_shape(ctx::GraphicsContext, rect::Rectangle)
    # use x, y, w, and h
end
function draw_shape(ctx::GraphicsContext, rect::Lasso)
    # use pts
end
```
where both `Rectangle` and `Lasso` are subtypes of `Shape`.